src/commons/sideContent/SideContentContestVoting.tsx

```diff
@@ -17,79 +17,205 @@ type StateProps = {
+            <div
+              className="item-container"
+              id="items"
+              onDragOver={handleDragOver}
+              onDragLeave={handleDragLeave}
+              onDragEnter={handleDragEnter}
+              onDrop={handleDrop}
+            >
+              {sortedContestEntries.map((contestEntry: ContestEntry, index) => (
+                <div
+                  className={classNames('item', { wrong: !isValid })}
+                  draggable={canSave}
+                  key={`item-${index + 1}`}
+                  id={`item-${index + 1}`}
+                  onDragStart={handleDragStart}
+                  onDragEnd={handleDragEnd(contestEntry)}
+                >
+                  <Card
+                    className={Classes.INTERACTIVE}
+                    elevation={Elevation.ONE}
+                    onClick={() =>
+                      handleContestEntryClick(
+                        contestEntry.submission_id,
+                        contestEntry.answer.code ?? ''
+                      )
+                    }
+                  >
+                    <Pre className="contestentry-entryid">{index + 1}</Pre>
+                  </Card>
+                </div>
+              ))}
+            </div>
+          </div>
         ) : (
           <div className="noResults">There are no eligible entries for voting found.</div>
         )}
       </div>
     ),
-    // determines when to re-render
     [
       isValid,
       canSave,
-      contestEntries,
+      sortedContestEntries,
       handleContestEntryClick,
-      handleVotingSubmissionChange,
-      maxScore,
-      minScore
+      handleDragEnd,
+      handleDragEnter,
+      handleDragLeave,
+      handleDragOver,
+      handleDrop,
+      tierBoard
     ]
   );
 
+  // Function for mapping legacy (and current) contest votes to the new tier system
+  const mapTier = (entryScore: number) => {
+    if (entryScore < 2) {
+      return (tierScore: number) => tierScore === 1;
+    }
+    if (entryScore >= 2 && entryScore < 4) {
+      return (tierScore: number) => tierScore === 2;
+    }
+    if (entryScore >= 4 && entryScore < 7) {
+      return (tierScore: number) => tierScore === 4;
+    }
+    if (entryScore >= 7 && entryScore < 10) {
+      return (tierScore: number) => tierScore === 7;
+    }
+    if (entryScore >= 10) {
+      return (tierScore: number) => tierScore === 10;
+    }
+    return (tierScore: number) => false;
+  };
+
+  // Renders contest entries in the correct tier using saved score upon initial render
+  useEffect(() => {
+    sortedContestEntries.forEach((entry, index) => {
+      if (entry.score !== undefined && entry.score !== null) {
+        const entryScore = entry.score as number;
+        const tierIndex = TIERS.findIndex(tier => mapTier(entryScore)(tier.score));
+        if (tierIndex !== -1) {
+          const tierElement = document.querySelector(
+            `#tier-${TIERS[tierIndex].name.toLowerCase()} .item-container`
```

Comment:
The index here is unnecessary.

```suggestion
        const tier = TIERS.find(tier => mapTier(entryScore)(tier.score));
        if (tier) {
          const tierElement = document.querySelector(
            `#tier-${tier.name.toLowerCase()} .item-container`
```